staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -305,10 +305,12 @@ func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEv
 			resyncPeriod = minimumResyncPeriod
 		}
 
-		if resyncPeriod < s.resyncCheckPeriod {
+		if resyncPeriod < s.resyncCheckPeriod || s.resyncCheckPeriod == 0 {
 			if s.started {
-				glog.Warningf("resyncPeriod %d is smaller than resyncCheckPeriod %d and the informer has already started. Changing it to %d", resyncPeriod, s.resyncCheckPeriod, s.resyncCheckPeriod)
-				resyncPeriod = s.resyncCheckPeriod
+				if s.resyncCheckPeriod != 0 {
+					glog.Warningf("resyncPeriod %d is smaller than resyncCheckPeriod %d and the informer has already started. Changing it to %d", resyncPeriod, s.resyncCheckPeriod, s.resyncCheckPeriod)
```

Comment:
Sorry to nit pick on this, but do you want to have a warning if the default is 0, the informer has started, and you try to add a nonzero resync event handler to it?

Reply:
Just saw your other comment

Reply:
I can add an explicit message here but as I mentioned above you will get a warning from the determineResyncPeriod() call that happens when it adds the handler. If you think having it explicitly here it's easy enough to add

Reply:
No, I wrote this before I saw your comment about the other warning

Reply:
I would prefer to make this more explicit and warn here + set resyncPeriod to zero. The logic now is not obvious.

Reply:
SGTM

Reply:
Ok .. will do...